internal/logger/logger.go

```diff
@@ -0,0 +1,130 @@
+package logger
```

Comment:
Alternatively, this is something off-the-shelf that might accomplish what you're trying to do and doesn't pull in additional dependencies: https://github.com/go-logr/logr

@grossjo for thoughts (and see prior comment)

Reply:
@cmlight I see your point. I'm also taking with @grossjo and @mlb7687 for alternative approaches. Will give an update soon.

Reply:
@cmlight I think your point about the utility code performing its own logic and its risk makes sense, I guess in that case log level needs to be set through whatever logger being used and methods like Info(), Warning(), etc in the logger should have log level checks implemented.

1) Regarding the conditional casting to the "standard" log package implementation, I think the idea is to be able to use this logger as default. If we only have interfaces, does that mean that we need to come up with a default logger to pass to the Logger interface?

2) I did look at https://github.com/go-logr/logr which seems like it's taking the similar approach of defining a generic logger interface. But I think it's a bit limited since it only supports Info, and Error (it argues that those are the only two levels which is needed https://dave.cheney.net/2015/11/05/lets-talk-about-logging). As an alternative have you seen this library https://github.com/Sirupsen/logrus ? It's not as light weight as logr but it is structured and supports multiple log levels. It also seems more easy to use and intuitive to me. It supports adding custom hooks, as testing as well.

3) I also found glog itself has four log levels which can be set through flags and can be enabled/disabled with `logtostderr=true/false`.
```
func init() {
    flag.Usage = usage
    flag.Set("logtostderr", "true")
    flag.Set("stderrthreshold", "WARNING")
    flag.Set("v", "2")
    // This is wa
    flag.Parse()
}

func main() {
    number_of_lines := 10
    for i := 0; i < number_of_lines; i++ {
        glog.V(2).Infof("LINE: %d", i)
        message := fmt.Sprintf("TEST LINE: %d", i)
        glog.Warning(message)
    }
    glog.Flush()
}
```

Reply:
Let me work more on the current change I'm making that should significantly reduce the need for there to be any log statements in the code to begin with.  Maybe anything that's left over can just be converted to the normal "log" package.  I'll circle back to you on this.

Reply:
My two cents, is we stick with this implementation.
But we update the documentation to reflect these facts:
1. clearly state about the secondary log level
2. this is a debug tool and not to be used all the time in production systems (one reason: printf style interface has performance impacts)

And then as we move forward on the project, we ensure that any log statements added are meaningful to debug the library and not just for development purposes.